### PR TITLE
[WIP] docs(aio):edit architecture doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
@StephenFluin @robwormald  Please check out proposed reorg/rewrite of architecture page - we should meet and talk about it.

### CODE REVIEW

File: aio/content/navigation.json [hunk 1]
```diff
@@ -127,7 +127,24 @@
         {
           "url": "guide/architecture",
```
Comment (gkalpak) [inline aio/content/navigation.json]: We have decided that it is bad UX to have URLs on expanding entries. This should be moved to child entry (e.g. "overview" or something).

**Comment by jbogarthyde** [comment]
Note that latest commits did not make it into the preview

**Comment by gkalpak** [comment]
> Note that latest commits did not make it into the preview

What do you mean, @jbogarthyde?
Which commits didn't make it?

**Comment by jbogarthyde** [comment]
Looks like it’s all there now. Took a long time.

**Review by StephenFluin — COMMENTED** [review]
LGTM

**Pull request closed without merging**